### DISCUSSION TITLE
How to Use Filter in Grid Block Table

### DISCUSSION THREAD

**Discussion by muksidin** [Budibase/budibase#11169] [Help] [answered]
Hello, i try to use grid block and then add component Dynamic Filter but the filter does not working. Is there a way to use filter in grid block.
Thanks team :)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @muksidin,

The Grid block doesn't support components interacting with it. As it has no data provider attached and no option to eject. It is a very isolated component. 

You can add filtering within the component itself. This can be used with other input fields as long as the grid is within the form.
<img width="1509" alt="Screenshot 2023-07-10 at 08 36 07" src="https://github.com/Budibase/budibase/assets/126772285/dc545099-a0a5-4106-85ab-aad8ba81d6e6">
<img width="1339" alt="Screenshot 2023-07-10 at 08 36 53" src="https://github.com/Budibase/budibase/assets/126772285/1027c9b4-69b9-4771-9dd9-d1dd14c6ce20">

Back to your question, as the grid doesn't use a `Data Provider`, the dynamic filter component can't be used with the new grid component due to it requiring it to be placed inside a data provider component.

I hope this makes sense and is helpful for you.

**Comment by muksidin** [comment]
Thanks @ConorWebb96 , very helpfull.
